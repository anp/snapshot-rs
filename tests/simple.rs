mod test {
    use serde_derive::{Deserialize, Serialize};
    use snapshot::snapshot;

    #[snapshot]
    fn simple_snapshot() -> i32 {
        let x = 1;
        x
    }

    #[cfg(test)]
    mod sub_test {
        use snapshot::snapshot;

        #[snapshot]
        fn simple_snapshot() -> String {
            "Nested".to_owned()
        }
    }

    #[derive(Debug, Deserialize, PartialEq, Serialize)]
    struct Lol {
        a: f32,
        x: i32,
        z: String,
    }

    #[snapshot]
    fn compound_snapshot() -> Lol {
        Lol {
            a: 1.0,
            x: 12,
            z: String::from("woowwowow"),
        }
    }
}
