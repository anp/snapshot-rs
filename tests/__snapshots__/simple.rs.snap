{
  "simple::test::compound_snapshot": {
    "file": "tests/simple.rs",
    "module_path": "simple::test",
    "test_function": "compound_snapshot",
    "recorded_value": {
      "a": 1.0,
      "x": 12,
      "z": "woowwowow"
    }
  },
  "simple::test::simple_snapshot": {
    "file": "tests/simple.rs",
    "module_path": "simple::test",
    "test_function": "simple_snapshot",
    "recorded_value": 1
  },
  "simple::test::sub_test::simple_snapshot": {
    "file": "tests/simple.rs",
    "module_path": "simple::test::sub_test",
    "test_function": "simple_snapshot",
    "recorded_value": "Nested"
  }
}